src/plots/plots.js

```diff
@@ -809,6 +809,16 @@ plots.purge = function(gd) {
     // remove modebar
     if(fullLayout._modeBar) fullLayout._modeBar.destroy();
 
+    if(fullLayout._plots) {
```

Comment:
... looks like `plots.purge` `delete`s `gd._fullLayout` .*just before* `Plotly.plot` is called (the latter of which then calls `supplyDefaults` which calls `cleanPlot`) so by the time `cleanPlot` is called, there's no longer information about _what_ to clean.

I'm in a bit of a conundrum as there's a big amount of statefulness and I'm concerned that a purported fix from me (e.g. *not* delete `gd._fullLayout` in `purge` so the info stays around) has unintended, undesirable consequences, besides the fact that it would allow proper working of the `_module.clean` loop. So I'd rather get input @etpinard before shoehorning in something that looks like a fix.